src/BizHawk.Emulation.Common/Interfaces/Services/ISettable.cs

```diff
@@ -69,46 +69,46 @@ public enum PutSettingsDirtyBits
 	/// </summary>
 	public class SettingsAdapter
 	{
-		public SettingsAdapter(IEmulator e)
+		public SettingsAdapter(IEmulator emulator)
 		{
-			_emu = e;
-
-			Type impl = e.GetType().GetInterfaces()
-				.FirstOrDefault(t => t.IsGenericType && t.GetGenericTypeDefinition() == typeof(ISettable<,>));
-			if (impl == null)
+			var settableType = emulator.ServiceProvider.AvailableServices

```

Comment:
@adelikat Does this look right?  It seems like such an obvious bug that I can't believe it's been this way for so long.

Reply:
Pretty sure it is an obvious bug, gone unnoticed since it was always the core class implementing the interface